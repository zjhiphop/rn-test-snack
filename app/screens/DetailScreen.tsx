import { observer } from "mobx-react-lite"
import React, { FC } from "react"
import { ViewStyle } from "react-native"
import { Screen, Text } from "../components"
import { AppStackScreenProps } from "../navigators"
// import { useNavigation } from "@react-navigation/native"
// import { useStores } from "../models"

interface DetailScreenProps extends AppStackScreenProps<"Detail"> {}

// STOP! READ ME FIRST!
// To fix the TS error below, you'll need to add the following things in your navigation config:
// - Add `Detail: undefined` to AppStackParamList
// - Import your screen, and add it to the stack:
//     `<Stack.Screen name="Detail" component={DetailScreen} />`
// Hint: Look for the üî•!

// REMOVE ME! ‚¨áÔ∏è This TS ignore will not be necessary after you've added the correct navigator param type
// @ts-ignore
export const DetailScreenComponent = observer(function DetailScreenComponent() {
  // Pull in one of our MST stores
  // const { someStore, anotherStore } = useStores()

  // Pull in navigation via hook
  // const navigation = useNavigation()
  return <Text text="detail" />
})

export const DetailScreen: FC<DetailScreenProps> = observer(function DetailScreen() {
  return (
    <Screen style={$root} preset="scroll">
      <DetailScreenComponent></DetailScreenComponent>
    </Screen>
  )
})

const $root: ViewStyle = {
  flex: 1,
}
