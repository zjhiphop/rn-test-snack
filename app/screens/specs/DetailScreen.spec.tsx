// https://reactnativetesting.io/component/testing/

import { render } from "@testing-library/react-native"
import React from "react"
import renderer from "react-test-renderer"
import { DetailScreenComponent } from "../DetailScreen"

describe("Detail", () => {
  it("renders DetailScreenComponent component", () => {
    const screen = render(<DetailScreenComponent></DetailScreenComponent>)
    expect(screen.getByText("detail")).toBeTruthy()
  })
})

describe("Renders Snapshot correctly", () => {
  const tree = renderer.create(<DetailScreenComponent></DetailScreenComponent>).toJSON()
  expect(tree).toMatchSnapshot()
})
