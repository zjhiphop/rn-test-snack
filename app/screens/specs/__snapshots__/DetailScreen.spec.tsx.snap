// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Text
  style={
    Array [
      Object {},
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "fontFamily": "spaceGroteskRegular",
        },
        Object {
          "color": "#191015",
        },
      ],
      undefined,
      undefined,
      undefined,
    ]
  }
>
  detail
</Text>
`;
